geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/conf/AbstractArgResolver.scala

```diff
@@ -0,0 +1,93 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.utils.conf
+
+import com.typesafe.config.ConfigParseOptions
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.annotation.tailrec
+
+/**
+ * Abstract trait for resolving sft/config parameters and handling errors around
+ * parsing.
+ */
+trait AbstractArgResolver[ReturnType, ParseMethodArgs] extends LazyLogging {
+
+  // Object for holding relevant error to return
+  class ErrorData {
```

Comment:
Is there a reason you can't use Exception here since Exception has message and causedBy?